apollo-router/src/plugins/authentication.rs

```diff
@@ -0,0 +1,944 @@
+                    let jwt_value_result =
+                        match request.router_request.headers().get(&my_config.header_name) {
+                            Some(value) => value.to_str(),
+                            None => {
+                                return failure_message(
+                                    request.context,
+                                    format!("Missing {} header", &my_config.header_name),
+                                    StatusCode::UNAUTHORIZED,
+                                );
+                            }
+                        };
+
+                    // If we find the header, but can't convert it to a string, let the client know
+                    let jwt_value_untrimmed = match jwt_value_result {
+                        Ok(value) => value,
+                        Err(_not_a_string_error) => {
+                            return failure_message(
+                                request.context,
+                                "configured header is not convertible to a string".to_string(),
+                                StatusCode::BAD_REQUEST,
+                            );
+                        }
+                    };
+
+                    // Let's trim out leading and trailing whitespace to be accommodating
+                    let jwt_value = jwt_value_untrimmed.trim();
+
+                    // Make sure the format of our message matches our expectations
+                    // Technically, the spec is case sensitive, but let's accept
+                    // case variations
+                    if !jwt_value
+                        .to_uppercase()
+                        .as_str()
+                        .starts_with(&my_config.header_value_prefix.to_uppercase())
+                    {
+                        return failure_message(
+                            request.context,
+                            format!(
+                                "Header Value: '{jwt_value_untrimmed}' is not correctly formatted. prefix should be '{}'",
+                                my_config.header_value_prefix
+                            ),
+                            StatusCode::BAD_REQUEST,
+                        );
+                    }
+
+                    // Split our string in (at most 2) sections.
+                    let jwt_parts: Vec<&str> = jwt_value.splitn(2, ' ').collect();
+                    if jwt_parts.len() != 2 {
+                        return failure_message(
+                            request.context,
+                            format!("Header Value: '{jwt_value}' is not correctly formatted. Missing JWT"),
+                            StatusCode::BAD_REQUEST,
+                        );
+                    }
+
+                    // Trim off any trailing white space (not valid in BASE64 encoding)
+                    let jwt = jwt_parts[1].trim_end();
```

Comment:
this was done already by the call to `trim` on line 271

Reply:
removed: 1bd8ed69f1d810adef32d3ad57d20d02328c0f6d